llmfoundry/utils/config_utils.py

```diff
@@ -16,14 +16,19 @@
 def pop_config(cfg: DictConfig,
                key: str,
                must_exist: bool = True,
-               default_value: Any = None) -> Any:
+               default_value: Any = None,
+               convert: bool = False) -> Any:
     """Pop a value from the main config file and return it.
 
     If the key does not exist, return the default_value or raise a RuntimeError
-    depending on the must_exist flag.
+    depending on the must_exist flag. If the convert flag is set to True, then
+    we will convert the value to a python object using OmegaConf.to_container.
     """
     value = cfg.pop(key, None)
-    if value is not None:
+    if value is not None and convert:
+        assert isinstance(value, DictConfig) or isinstance(value, ListConfig)
```

Comment:
this should be a valueerror not an assert. generally speaking, errors should be used when it is a message for the user, and assert is used when it is just to satisfy pyright or make an invariant clear.

Reply:
ya good point let me fix